### PR TITLE
feat: added genesis function for revenue distribution

### PR DESCRIPTION
**Pull request by gyuguen**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,107 +3,37 @@ module github.com/medibloc/panacea-core/v2
 go 1.15
 
 require (
```
Comment (gyuguen) [inline go.mod:5]: Well, in my idea, all the deleted dependencies are not needed.
I think if there are no problems, we can continue as it is.

**Comment by gyuguen** [comment]
I applied what we discussed offline here.

**Review by inchori — APPROVED** [review]
LGTM 👍

File: x/datapool/genesis.go [hunk 2]
```diff
@@ -23,6 +23,13 @@ func InitGenesis(ctx sdk.Context, k keeper.Keeper, genState types.GenesisState)
 		k.SetPool(ctx, &pool)
 	}
 
+	k.SetInstantRevenueDistribute(ctx, &genState.InstantRevenueDistribute)
+
+	for _, history := range genState.SalesHistories {
+		key := types.GetKeyPrefixSalesHistory(history.PoolId, history.Round, history.SellerAddress)
+		k.SetSalesHistoryByKey(ctx, key, history)
```
Comment (hansol-medi) [inline x/datapool/genesis.go:30]: I think you can use `k.SetSalesHistory()` itself.
```suggestion
		k.SetSalesHistory(ctx, history)
```

File: x/datapool/keeper/pool_distribute.go [hunk 3]
```diff
@@ -32,6 +32,11 @@ func (k Keeper) SetSalesHistory(ctx sdk.Context, salesHistory *types.SalesHistor
 		salesHistory.Round,
 		salesHistory.SellerAddress,
 	)
+	k.SetSalesHistoryByKey(ctx, key, salesHistory)
+}
+
+// SetSalesHistoryByKey stores sales history.
+func (k Keeper) SetSalesHistoryByKey(ctx sdk.Context, key []byte, salesHistory *types.SalesHistory) {
```
Comment (hansol-medi) [inline x/datapool/keeper/pool_distribute.go:39]: I just wonder why you made this function splitted which stores history by key. Is there any specific reason?!

**Review by hansol-medi — APPROVED** [review]
LGTM

**Review by youngjoon-lee — APPROVED** [review]
good

File: x/datapool/genesis.go (hunk 2 above)
Comment (gyuguen) [reply x/datapool/genesis.go]: Thank you!

File: x/datapool/keeper/pool_distribute.go (hunk 3 above)
Comment (gyuguen) [reply x/datapool/keeper/pool_distribute.go]: It is a residue created while editing the code.
I removed this function. :)

**Pull request merged**